Authentiication card completed

Mawadddah opened the pull request.

Mawadddah committed:
Authentiication card completed

tekul commented on client/src/index.js:
```diff
@@ -2,11 +2,18 @@ import React from "react";
 import ReactDOM from "react-dom";
 import "./index.css";
 import Home from "./components/Home";
-import "semantic-ui-css/semantic.min.css";
```
This line is deleted here and added again at line 8. Try not to move lines about unnecessarily as it makes it harder to read the commit changes that are relevant to the task.

Mawadddah replied:
Noted

tekul reviewed (changes requested):
Looks good and I tested it locally and was able to log in so it all works as expected :+1: . Could we use semantic UI for the login form components though, so that the style matches the rest of the site?

bonboh commented on client/src/api/auth.js:
```diff
@@ -0,0 +1,14 @@
+export const signApi = (email, password) => {
```
To me it's not so clear what a function called `signApi` does. We write code for other developers, so naming becomes quite important! What does this function do? It logs in a user, so we could name it `loginUser` or `signInUser` to be more easily understood

bonboh commented on client/src/api/auth.js:
```diff
@@ -0,0 +1,14 @@
+export const signApi = (email, password) => {
+  const opts = {
```
Various code style guides prefer full words rather than abbreviations, so we could rename `opts` to `options`

bonboh commented on client/src/api/auth.js:
```diff
@@ -0,0 +1,14 @@
+  };
+  return fetch("/auth/login", opts)
+    .then(res => res.json())
+    .then(data => {
+      return data;
+    });
```
```suggestion
```
Since the second `then` outputs the same input, we can remove it since it's unnecessary

bonboh commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,55 @@
+    signApi(email, password)
+      .then(data => {
+        const token = data.token;
+        console.log(token);
```
It's generally good practise to remove all `console.log`s from the frontend, so let's remove it here

bonboh commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,55 @@
+    return <div>You are already logged in.</div>;
+  } else
+    return (
+      <div>
```
It's good practise to wrap inputs and submit buttons in a `form`, and to provide a label for every input, so let's do that. Also, let's use Semantic UI components! 

Here's an example of creating a form with Semantic UI: https://react.semantic-ui.com/collections/form/#types-form

bonboh replied:
Since this `div` isn't needed, let's remove it!

bonboh commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,55 @@
+          placeholder="password"
+        />
+        <button onClick={signIn}>Sign in</button>
+        {error ? <div>Wrong information. Try again</div> : null}
```
We should aim to write semantic HTML as much as we can. Since this is a sentence, it is more semantic to have it in a paragraph `p` tag so let's change it.

Since this is a user-facing message, we could also provide a more gentle message. Maybe something like `The login details are not correct. Please try again`?

Mawadddah replied:
Thank you I just changed it :)

bonboh reviewed (changes requested):
Great work @AaishaMohamad! I've left a few comments for you

Mawadddah committed:
Merge branch 'master' of https://github.com/CodeYourFuture/scot-min-imece into Aaisha/authentiication

Mawadddah committed:
Authentication card 2

tekul reviewed (changes requested):
Try to use semantic-ui for the login form UI (input, buttons etc). You can use this example here as a guide https://react.semantic-ui.com/layouts/login/

bonboh commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import "semantic-ui-css/semantic.min.css";
```
We only need to import the Semantic UI CSS once in our project and we're doing that in `client/src/index.js`, so let's remove this import

bonboh commented on client/src/components/Login.js:
```diff
@@ -0,0 +1,66 @@
+  } else
+    return (
+      <div>
+        <form class="ui form">
+          <div class="field">
+            <label>Email</label>
+            <input
+              onChange={handleChange}
+              name="email"
+              type="email"
+              placeholder="Example@email.com"
+            />
+          </div>
+          <div class="field">
+            <label>Password</label>
+            <input
+              onChange={handleChange}
+              name="password"
+              type="password"
+              placeholder=" Enter password"
+            />
+          </div>
+          <button onClick={signIn} class="ui button">
+            Sign in
+          </button>
+          {error ? (
+            <div>The login details are not correct. Please try again</div>
+          ) : null}
+        </form>
```
Nice job updating this to use a form, labels, and Semantic UI styling! However, in this project we want to use Semantic UI React components so instead we want to import them into this file and use them like so:

```
import { Button, Container, Form, Input } from 'semantic-ui-react';

<Form onSubmit={signIn}>
  <Form.Field>
    <label for="email">Email</label>
    <Input
      id="email"
      onChange={handleChange}
      name="email"
      type="email"
      placeholder="Example@email.com"
    />
  </Form.Field>
  <Form.Field>
    <label for="password">Password</label>
    <Input
      id="password"
      onChange={handleChange}
      name="password"
      type="password"
      placeholder=" Enter password"
    />
  </Form.Field>
  <Button>Sign in</Button>
  {error ? (
    <div>The login details are not correct. Please try again</div>
  ) : null}
</Form>
```

There's a few things to note:

- Using Semantic UI means we don't need to add Semantic UI class names because the components have them inside!
- Labels and inputs need to be connected using `for` and `id` (see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label for documentation on this)—this is very important for web accessibility, which is being discussed more in the frontend community. This also means that by clicking on the label (Email or Password), the input is automatically selected
- When using forms, pressing `Enter` when in an input or clicking on a button will submit the form. Therefore, we need to move the `signIn` function to the form `onSubmit` event.

Hope that all makes sense!

bonboh reviewed (changes requested):
Nice update @AaishaMohamad! I've left a few comments, and once we update to using Semantic UI then we should be good 👍

Mawadddah committed:
Used semantic-ui for login page

Mawadddah committed:
Add name attribute to login input fields

These attributes were missing after switching to semantic-ui, so the login data being sent to the server was empty.

tekul commented:
Merged manually
